File: app/src/main/java/bapspatil/captainchef/network/BakingAPI.java
package bapspatil.captainchef.network;

import java.util.List;

import bapspatil.captainchef.model.FoodItem;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;

/**
 * Created by bapspatil
 */

public interface BakingAPI {

    @GET("resources/baking.json")
    Call<List<FoodItem>> getFoodItems();

    Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("https://raw.githubusercontent.com/bapspatil/CaptainChef/master/")
            .addConverterFactory(GsonConverterFactory.create())
            .build();
}

File: app/src/main/java/bapspatil/captainchef/network/BakingAPI.kt
package bapspatil.captainchef.network

import bapspatil.captainchef.model.FoodItem
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import java.util.*

/**
 * Created by bapspatil
 */

interface BakingAPI {

    @get:GET("resources/baking.json") val foodItems: Call<ArrayList<FoodItem>>

    companion object {

        const val BASE_URL = "https://raw.githubusercontent.com/bapspatil/CaptainChef/master/"

        val retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
    }
}
